apiserver/logtransfer.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2015 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package apiserver
+
+import (
+	"io"
+	"net/http"
+
+	"github.com/juju/errors"
+	"github.com/juju/loggo"
+	"github.com/juju/version"
+
+	"github.com/juju/juju/apiserver/params"
+	"github.com/juju/juju/state"
+)
+
+type migrationLoggingStrategy struct {
+	ctxt       httpContext
+	st         *state.State
+	version    version.Number
+	filePrefix string
+	dbLogger   *state.DbLogger
+	fileLogger io.Writer
+}
+
+// Authenticate checks that the user is a controller superuser and
+// that the requested model is migrating. Part of LoggingStrategy.
+func (s *migrationLoggingStrategy) Authenticate(req *http.Request) error {
+	st, err := s.ctxt.stateForMigration(req)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	// Here the log messages are expected to be coming from another
+	// Juju controller, so the version number provided should be the
+	// Juju version of the source controller.
+	ver, err := jujuClientVersionFromReq(req)
```

Comment:
As discussed in a hangout yesterday, storing the version per log message is kinda dumb and we shouldn't persist with that mistake with migrations. Sending the client's Juju version on connect is potentially useful though so let's keep doing that.